controllers/datadogagent/agent_test.go

```diff
@@ -825,6 +825,20 @@ func defaultPodSpec() corev1.PodSpec {
 }
 
 func defaultSystemProbePodSpec() corev1.PodSpec {
+	agentWithSystemProbeVolumeMounts := []corev1.VolumeMount{}
+	agentWithSystemProbeVolumeMounts = append(agentWithSystemProbeVolumeMounts, defaultMountVolume()...)
+	agentWithSystemProbeVolumeMounts = append(agentWithSystemProbeVolumeMounts, []corev1.VolumeMount{
+		{
+			Name:      "sysprobe-socket-dir",
+			ReadOnly:  true,
+			MountPath: "/opt/datadog-agent/run",
+		},
+		{
+			Name:      "system-probe-config",
+			MountPath: "/etc/datadog-agent/system-probe.yaml",
+			SubPath:   "system-probe.yaml",
+		},
```

Comment:
does it mean that before we didn't mount the config map that contains the system-probe.yaml configuration?

Reply:
Exactly

Reply:
https://github.com/DataDog/datadog-operator/blob/15e9acebe3d7340f1954c745dbb3047f3ba70a49/controllers/datadogagent/utils.go#L838

Reply:
The thing is that, before the `oom_kill` and `tcp_queue_length` checks, `system-probe` talked only to the `process-agent`.
So, it was not needed to access `system-probe.yaml` from the core agent.

When `oom_kill` and `tcp_queue_length` checks were introduced, it initially worked without `system-probe.yaml` because the core agent was looking for the `system-probe` socket at its default path. (And as we are inside a container, there was no value to make this path configurable.)
But following some code change, I think that the core agent now needs to have access to `system-probe.yaml` to have the socket path.

Reply:
https://github.com/DataDog/datadog-operator/blob/15e9acebe3d7340f1954c745dbb3047f3ba70a49/controllers/datadogagent/utils.go#L838

Reply:
BTW running `agent check tcp_queue_length ` still does not work, wonder if related?

Reply:
should we create an `emptyDir` to share the socket between the agents  that need it?

Reply:
But the metrics are showing up in the app ?

Reply:
Yes status is fine too

Reply:
i think you have to do something like `agent check tcp_queue_length -c /etc/datadog-agent/system-probe.yaml`